packages/wrapped-keys/src/lib/wrapped-keys.ts

```diff
@@ -0,0 +1,216 @@
+  const allowPkpAddressToDecrypt = getPkpAccessControlCondition(pkpAddress);
+
+  try {
+    const response = await fetch(ENCRYPTED_PRIVATE_KEY_ENDPOINT, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        pkpsessionsig: JSON.stringify(firstSessionSig),
+      },
+    });
+
+    if (!response.ok) {
+      const errorBody = await response.text();
+      logError(
+        `Could not fetch the encrypted key due to the error: ${errorBody}`
+      );
+
+      throw new Error(errorBody);
+    }
+
+    const exportedPrivateKeyData: ExportPrivateKeyResponse =
+      await response.json();
+
+    const decryptedPrivateKey = await decryptToString(
+      {
+        accessControlConditions: allowPkpAddressToDecrypt,
+        chain: CHAIN_ETHEREUM,
+        ciphertext: exportedPrivateKeyData.ciphertext,
+        dataToEncryptHash: exportedPrivateKeyData.dataToEncryptHash,
+        sessionSigs: pkpSessionSigs,
+      },
+      litNodeClient
+    );
+
+    return decryptedPrivateKey;
+  } catch (error) {
+    const errorMessage = `There was a problem fetching from the database: ${error}`;
+    console.error(errorMessage);
+
+    throw new Error(errorMessage);
+  }
+}
+
+export async function signWithEncryptedKey<T = LitMessage | LitTransaction>({
+  pkpSessionSigs,
+  litActionCode,
+  unsignedTransaction,
+  broadcast,
+  litNodeClient,
+}: SignWithEncryptedKeyParams<T>): Promise<string> {
+  const firstSessionSig = getFirstSessionSig(pkpSessionSigs);
+
+  let pkpAddress: string, ciphertext: string, dataToEncryptHash: string;
+
+  try {
+    const response = await fetch(ENCRYPTED_PRIVATE_KEY_ENDPOINT, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        pkpsessionsig: JSON.stringify(firstSessionSig),
+      },
+    });
+
+    if (!response.ok) {
+      const errorBody = await response.text();
+      logError(
+        `Could not fetch the encrypted key due to the error: ${errorBody}`
+      );
+
+      throw new Error(errorBody);
+    }
+
+    const exportedPrivateKeyData: ExportPrivateKeyResponse =
+      await response.json();
+
+    ({ pkpAddress, ciphertext, dataToEncryptHash } = exportedPrivateKeyData);
+  } catch (error) {
+    const errorMessage = `There was a problem fetching from the database: ${error}`;
+    console.error(errorMessage);
+
+    throw new Error(errorMessage);
+  }
+
+  let result;
+  try {
+    result = await litNodeClient.executeJs({
+      sessionSigs: pkpSessionSigs,
+      code: litActionCode,
+      jsParams: {
+        pkpAddress,
+        ciphertext,
+        dataToEncryptHash,
+        unsignedTransaction,
+        broadcast,
+        accessControlConditions: getPkpAccessControlCondition(pkpAddress),
+      },
+    });
+  } catch (err) {
+    throw new Error(
+      `Lit Action threw an unexpected error: ${JSON.stringify(err)}`
+    );
+  }
+
+  console.log(`Lit Action result: ${JSON.stringify(result)}`);
```

Comment:
Are we guaranteed this is always json?

Reply:
This is the Lit Action result which is always a JSON